sync3/lists.go

```diff
@@ -222,6 +222,34 @@ func (s *InternalRequestLists) ListsByVisibleRoomIDs(muxedReqLists map[string]Re
 	return listsByRoomIDs
 }
 
+// Visible determines if a single room is currently visible in the given set of lists.
+func (s *InternalRequestLists) Visible(roomID string, muxedReqLists map[string]RequestList) bool {
```

Comment:
Honestly I would remove this and not have this functionality:
```
// Don't bother queuing the event if the room isn't visible to the user.
```

We have per room queues in txn_id_waiter, just feed them through ignoring the visibility and let conn state handle it. It feels like a smell to have txn id waiter know or care about visible rooms.

Reply:
Was basing this on

> It needs some information from the conn ([roomSubscriptions](https://github.com/matrix-org/sliding-sync/blob/main/sync3/handler/connstate.go?rgh-link-date=2023-06-13T13%3A02%3A26Z#L31) to see if there is a room subscription for this event? That doesn't cover list windows though...)

from https://github.com/matrix-org/sliding-sync/issues/3#issuecomment-1589270205.

Happy to rip it out though; agree that it's smelly.